app/services/migration/user_migrator.rb

```diff
@@ -0,0 +1,117 @@
+module Migration
+  class UserMigrator
+    attr_reader :ecf_user
+
+    def initialize(ecf_user)
+      @ecf_user = ecf_user
+    end
+
+    def find_initialize_or_merge_user
+      if user_does_not_exist?
```

Comment:
Can we make these methods clear on which source we're looking at, so `ecf_user_does_not_exist?` and `both_ecf_ids_return_same_user?` etc